frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,471 @@
+                {status}: {(installObj as ClusterServiceVersionKind)?.status?.message || ''}
+              </Alert>
+            )}
+            {!loading && (
+              <Card>
+                <CardBody>
+                  <div className="co-operator-install-page__pkg-indicator">
+                    <div>{getCSVLogo()}</div>
+                    <div>{getIndicator()}</div>
+                  </div>
+                </CardBody>
+              </Card>
+            )}
+            {!loading && (
+              <Card>
+                <CardBody>{getInstallMessage()}</CardBody>
+              </Card>
+            )}
+          </div>
+        </Bullseye>
+      </div>
+      {!loading && isApprovalNeeded && (
+        <InstallPlanPreview obj={installObj as InstallPlanKind} hideApprovalBlock />
+      )}
+    </>
+  );
+};
+
+export const OperatorInstallPage: React.FC<OperatorInstallPageProps> = (props) => {
+  const { pkgNameWithVersion, targetNamespace } = props;
+  const namespace = targetNamespace;
+  const pkg = new URLSearchParams(window.location.search).get('pkg');
+  const installPageResources = [
+    {
+      kind: referenceForModel(ClusterServiceVersionModel),
+      namespaced: true,
+      fieldSelector: `metadata.name=${pkgNameWithVersion}`,
+      isList: true,
+      namespace,
+      prop: 'clusterServiceVersions',
+    },
+    {
+      kind: referenceForModel(SubscriptionModel),
+      namespaced: true,
+      fieldSelector: `metadata.name=${pkg}`,
+      isList: true,
+      namespace,
+      optional: true,
+      prop: 'subscription',
+    },
+    {
+      kind: referenceForModel(InstallPlanModel),
+      prop: 'installPlan',
+      namespaced: true,
+      namespace,
+      isList: true,
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PackageManifestModel),
+      namespace: new URLSearchParams(window.location.search).get('catalogNamespace'),
+      fieldSelector: `metadata.name=${pkg}`,
+      selector: {
+        matchLabels: {
+          catalog: new URLSearchParams(window.location.search).get('catalog'),
+        },
+      },
+      prop: 'packageManifest',
+    },
+  ];
+
+  return (
+    <Firehose resources={installPageResources}>
+      <OperatorInstallStatusPage {...props} />
+    </Firehose>
+  );
+};
+
+export type OperatorInstallPageProps = {
+  activeNamespace?: string;
+  targetNamespace: string;
+  match?: match;
+  pkgNameWithVersion: string;
+  resourceDescriptions: CRDDescription[];
+  resources?: FirehoseResourcesResult;
+};
+
+type InstallPageMessageProps = {
+  namespace: string;
+  className?: string;
+  obj?: ClusterServiceVersionKind | InstallPlanKind | SubscriptionKind;
+  csvName?: string;
+  approve?: () => void;
+};
+
+type InstallPageMessageButtonProps = {
+  namespace?: string;
+  className?: string;
+  obj?: ClusterServiceVersionKind | InstallPlanKind | SubscriptionKind;
+  disabled?: boolean;
```

Comment:
Are all of these props really optional?

Reply:
That prop is used for all the various Message Components, InstallSucceed, InstallFail, InstallWithCRD.  Some don't need items so I made them optional.  `obj` is used by all of them so I made that required.

Reply:
> Just like the messages, the button components vary slightly but have similar needs so I used a common props structure. I can break them out if you'd prefer, but yeah I needed to make them all optional.

If they the props are not exactly the same (including requiredness), it would be better to have separate types. I'm fine with a single type if the props really are identical for each one. (Same for messages.)